src/core/agent/services/connectionService.ts

```diff
@@ -224,6 +224,11 @@ class ConnectionService extends AgentService {
     for (const note of notes) {
       this.connectionNoteStorage.deleteById(note.id);
     }
+    const historyItems =
+      await this.ipexMessageStorage.getIpexMessageMetadataByConnectionId(id);
+    for (const historyItem of historyItems) {
+      this.ipexMessageStorage.deleteIpexMessageMetadata(historyItem.id);
```

Comment:
We need to `await` this to know everything has been deleted fully. Same above for connection notes actually. If we want to do it in parallel we can use promise.all